src/codec_svt.c

```diff
@@ -199,7 +208,7 @@ static avifResult svtCodecEncodeImage(avifCodec * codec,
     input_buffer->pts = 0;
 
     EbAv1PictureType frame_type = EB_AV1_INVALID_PICTURE;
-    if (addImageFlags & AVIF_ADD_IMAGE_FLAG_FORCE_KEYFRAME) {
+    if ((addImageFlags & AVIF_ADD_IMAGE_FLAG_FORCE_KEYFRAME) || (encoder->keyframeInterval == 1)) {
```

Comment:
@hassount Hassene:

Could you review our changes in this file? We want to set SVT-AV1's equivalent of libaom's `kf_max_dist` to implement the maximum keyframe interval.

We also have a question: the libavif encoder has an `AVIF_ADD_IMAGE_FLAG_FORCE_KEYFRAME` flag to force a given animation frame to be a keyframe. We implemented that by setting `input_buffer->pic_type` to `EB_AV1_KEY_PICTURE` here, but that doesn't work. Do you know why?

Reply:
@wantehchang 
I am not sure I understand what we're trying to do with this:
>Could you review our changes in this file? We want to set SVT-AV1's equivalent of libaom's kf_max_dist to implement the maximum keyframe interval.

In general, SVT has only one parameter for the key-frame setting and lines 164-165 seem to be the right way of setting it. that said to set 
`else if (encoder->keyframeInterval == 1) {`
`svt_config->intra_period_length =` should be just set to `0`.

Now as for forcing the key frames, I believe you'd need to set both `force_key_frames` set to 1 (once for the sequence) and `input_buffer->pic_type to EB_AV1_KEY_PICTURE`

Reply:
> @wantehchang I am not sure I understand what we're trying to do with this:
> 
> > Could you review our changes in this file? We want to set SVT-AV1's equivalent of libaom's kf_max_dist to implement the maximum keyframe interval.
> 
> In general, SVT has only one parameter for the key-frame setting and lines 164-165 seem to be the right way of setting it. that said to set `else if (encoder->keyframeInterval == 1) {` `svt_config->intra_period_length =` should be just set to `0`.
> 

i tried setting `svt_config->intra_period_length = 0` but it did not work as intended. (i.e.) the expectation is that all the frames will be keyframes in that case. but the output i got was the first frame was keyframes and subsequent frames were not.

any idea what i am missing?

> Now as for forcing the key frames, I believe you'd need to set both `force_key_frames` set to 1 (once for the sequence) and `input_buffer->pic_type to EB_AV1_KEY_PICTURE`

this is what this PR is now doing for the "all frames are key frames case".

Reply:
@1480c1 did we change the API to remove the +1 for intra_period ?
@vigneshvg can you try svt_config->intra_period_length = 1 instead ?

Reply:
> @vigneshvg can you try svt_config->intra_period_length = 1 instead ?

yes, but that just outputs a keyframe every other frame. in this case, we want every frame to be a keyframe.

Reply:
this is weird then. either 0 or 1 should work @1480c1 any thoughts on this ?

Reply:
@hassount @1480c1 I filed https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/2076 about the intra_period_length == 0 issue discussed above.